ui/warrior/inputs.ts

```diff
@@ -0,0 +1,391 @@
+
+// Configuration for spec-specific UI elements on the settings tab.
+// These don't need to be in a separate file but it keeps things cleaner.
+
+export const Recklessness = {
+	id: ActionId.fromSpellId(1719),
+	states: 2,
+	extraCssClasses: [
+		'warrior-recklessness-picker',
+	],
+	changedEvent: (player: Player<Spec.SpecWarrior>) => player.specOptionsChangeEmitter,
+	getValue: (player: Player<Spec.SpecWarrior>) => player.getSpecOptions().recklessness,
+	setValue: (eventID: EventID, player: Player<Spec.SpecWarrior>, newValue: boolean) => {
+		const newOptions = player.getSpecOptions();
+		newOptions.recklessness = newValue
+		player.setSpecOptions(eventID, newOptions);
+	},
+};
+
+export const WarriorRotationConfig = {
+	inputs: [
+		{
+			type: 'enum' as const,
+			getModObject: (simUI: IndividualSimUI<any>) => simUI,
+			config: {
+				extraCssClasses: [
+					'rotation-type-enum-picker',
+				],
+				label: 'Spec',
+				labelTooltip: 'Switches between spec rotation settings. Will also update talents to defaults for the selected spec.',
+				values: [
+					{
+						name: 'Arms Slam', value: RotationType.ArmsSlam,
+					},
+					{
+						name: 'Arms DW', value: RotationType.ArmsDW,
+					},
+					{
+						name: 'Fury', value: RotationType.Fury,
+					},
+				],
+				changedEvent: (simUI: IndividualSimUI<Spec.SpecWarrior>) => simUI.player.rotationChangeEmitter,
+				getValue: (simUI: IndividualSimUI<Spec.SpecWarrior>) => simUI.player.getRotation().type,
+				setValue: (eventID: EventID, simUI: IndividualSimUI<Spec.SpecWarrior>, newValue: number) => {
+					const newRotation = simUI.player.getRotation();
+					newRotation.type = newValue;
+
+					TypedEvent.freezeAllAndDo(() => {
+						if (newRotation.type == RotationType.Fury) {
+							simUI.player.setTalentsString(eventID, Presets.FuryTalents.data);
+							if (!newRotation.fury) {
+								newRotation.fury = FuryRotation.clone(Presets.DefaultFuryRotation.fury!);
+							}
+						} else if (newRotation.type == RotationType.ArmsSlam) {
+							simUI.player.setTalentsString(eventID, Presets.ArmsSlamTalents.data);
+							if (!newRotation.armsSlam) {
+								newRotation.armsSlam = ArmsSlamRotation.clone(Presets.DefaultArmsSlamRotation.armsSlam!);
+							}
+						} else {
+							simUI.player.setTalentsString(eventID, Presets.ArmsDWTalents.data);
+							if (!newRotation.armsDw) {
+								newRotation.armsDw = ArmsDWRotation.clone(Presets.DefaultArmsDWRotation.armsDw!);
+							}
+						}
+
+						simUI.player.setRotation(eventID, newRotation);
+					});
+
+					simUI.recomputeSettingsLayout();
+				},
+			},
+		},
+		// ********************************************************
+		//                       FURY INPUTS
+		// ********************************************************
+		{
+			type: 'enum' as const,
+			getModObject: (simUI: IndividualSimUI<any>) => simUI.player,
+			config: {
+				extraCssClasses: [
+					'rotation-type-enum-picker',
+				],
+				label: 'Primary Instant',
```

Comment:
A tooltip here might be a good idea, just to clarify what 'Primary Instant' does exactly. I assume something like, 'Highest priority ability to use. The other one will still be used when available and the primary is on CD.'

Reply:
Added tooltip on 2dc332d.